### PR TITLE
Introduce ceil division helper function.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libsolidity/codegen/YulUtilFunctions.cpp [hunk 1]
```diff
@@ -541,6 +541,21 @@ string YulUtilFunctions::roundUpFunction()
...
+	return m_functionCollector.createFunction(functionName, [&]() {
+		return
+			Whiskers(R"(
+			function <functionName>(value) -> result {
+				result := div(add(value, 31), 32)
+			}
+			)")
+			("functionName", functionName)
+			.render();
+	});
```
Comment (mijovic) [inline libsolidity/codegen/YulUtilFunctions.cpp]: Maybe using new interface, but not mandatory
```suggestion
	return m_functionCollector.createFunction(functionName, [&](vector<string>& _args, vector<string>& _returnValues) {
		_args = {"value"};
		_returnValues = {"result"};
		return
			Whiskers(R"(
				result := div(add(value, 31), 32)
			)")
			.render();
	});
```

File: libsolidity/codegen/YulUtilFunctions.h [hunk 2]
```diff
@@ -126,9 +126,16 @@ class YulUtilFunctions
...
 	/// @returns the name of a function that rounds its input to the next multiple
 	/// of 32 or the input if it is a multiple of 32.
+	/// Ignores overflow.
 	/// signature: (value) -> result
 	std::string roundUpFunction();
 
+	/// @returns the name of a function that divides by 32 and rounds up during the division.
+	/// In other words, on input x it returns the smallest y such that y * 32 >= x.
+	/// Ignores overflow.
+	/// signature: (x) -> y
```
Comment (mijovic) [inline libsolidity/codegen/YulUtilFunctions.h:136]: Align with implementation?
```suggestion
	/// signature: (value) -> result
```

**Review by mijovic — COMMENTED** [review]
Looks good, only two small things. Also, it is optional using new interface for create function. It doesn't duplicate function name, but adds additional parameters, so not sure if it is better in this case

File: libsolidity/codegen/YulUtilFunctions.cpp [hunk 3]
```diff
@@ -1257,14 +1272,15 @@ string YulUtilFunctions::cleanUpDynamicByteArrayEndSlotsFunction(ArrayType const
...
-				let deleteStart := add(dataArea, div(<roundUp>(startIndex), 32))
+				let deleteStart := add(dataArea, <div32Ceil>(startIndex))
 				// If we are clearing array to be short byte array, we want to clear only data starting from array data area.
 				if lt(startIndex, 32) { deleteStart := dataArea }
-				<clearStorageRange>(deleteStart, add(dataArea, div(add(len, 31), 32)))
+				<clearStorageRange>(deleteStart, add(dataArea, <div32Ceil>(len)))
 			}
 		)")
 		("dataLocation", arrayDataAreaFunction(_type))
 		("roundUp", roundUpFunction())
```
Comment (mijovic) [inline libsolidity/codegen/YulUtilFunctions.cpp]: You need to remove `roundUp` here, as it is not used
```suggestion
```

File: libsolidity/codegen/YulUtilFunctions.h (hunk 2 above)
Comment (chriseth) [reply libsolidity/codegen/YulUtilFunctions.h:136]: I used x and y because this is easier to read in the text description above.

**Review by mijovic — APPROVED** [review]

**Pull request merged**